### DISCUSSION TITLE
Using `range` in template on a db().all() variable

### DISCUSSION THREAD

**Discussion by inventor96** [pocketbase/pocketbase#3419] [Q&A] [answered]
I'm trying to populate `<option>` elements for a `<select>` element in a template. The application is currently hosted on pockethost.io. My `pb_hooks/main.pb.js` looks like this:

```js
routerAdd("GET", "/login", (c) => {
	const result = arrayOf(new DynamicModel({
		"id": "",
		"name": "",
	}))

	$app.dao().db()
		.select("id", "name")
		.from("stakes")
		.orderBy("name")
		.all(result)

	const html = $template.loadFiles(
		`${__hooks}/views/base.html`,
		`${__hooks}/views/login.html`,
	).render({
		"stakes": result,
	})

	return c.html(200, html)
})
```

And here's the relevant part of `login.html`:
```html
<select class="form-select" id="stake" name="stake">
	{{range .stakes}}
	<option value="{{.id}}">{{.name}}</option>
	{{end}}
</select>
```

When accessing the /login URL, I just get a generic `{"code":400,"message":"Something went wrong while processing your request.","data":{}}`. Being new to pocketbase, I've tried a few shots in the dark on other ways to try the templating, but to no avail. If I simplify the options to just `<option>{{.}}</option>`, then I get text that includes the ID and name, but obviously this is not the way I want it to work.

Could someone point me in the right direction on how to do this?

**Comment by inventor96** [comment, marked as answer]
Looks like I asked the question too soon. I changed the JS code to include a try-catch like this:
```js
var html;
try {
	html = $template.loadFiles(
		`${__hooks}/views/base.html`,
		`${__hooks}/views/login.html`,
	).render({
		"stakes": result,
	})
} catch (error) {
	return c.html(500, error)
}

return c.html(200, html)
```

Which then gave me an error that hinted at the first letter of the property being capitalized. Sure enough, after changing the respective part of the template to look like `<option value="{{.Id}}">{{.Name}}</option>`, it worked as expected. I'm sure this was a noob problem, but figured I'd share the solution here for others that might encounter this issue.

**Reply by ganigeorgiev** [reply]
Yes, the above solution is correct.

This is a side-effect from the go->js->go template resolution (_the syntax in the templates always expect the Go formatting; it is not very intuitive and I'll think a little more on it if it can be "normalized" or at least add a note in the documentation_).

If `result` was a simple array with plain objects then the above example should work just fine because `id` and `name` would have been simple string keys in each object.

But that's not the case with the `DynamicModel` because it is a wrapped Go struct, meaning that under the hood in the Go code its fields are capitalized to denote that are public.

_Side-note: you can also get more detailed errors in the console if you start the executable with `--debug`._
